WordPress/gutenberg
# Target Hints: Add missing param sanitization

## What?


This fixes a serious bug in the `Gutenberg_REST_Server` class which is intended for 6.7 merge. This bug causes fatal errors for plugins like Web Stories.

#64504 updated `WP_REST_Sever` to add the `targetHints` property. Like `WP_REST_Server::dispatch()` it uses `match_request_to_handler()` to get the correct handler and then "sends" the correct headers.

The problem is that it does not sanitize the request params it gathered from `WP_REST_Request::from_url()` and `match_request_to_handler()`. So if the controller in question expects an `int`, it will get a `string` and if it has strict typing, that leads to fatal errors.

**Note:** This might warrant a 19.2.1 release.

## Why?


Bugs are bad!

## How?


Fixes the bug by validating and sanitizing the params.

I don't have the capacity right now to add tests, so if someone else can do it, that would be great.

## Testing Instructions





1. Install the Web Stories plugin
2. Open the Web Stories editor
3. See (or not see) a fatal error

### Testing Instructions for Keyboard


N/A

## Screenshots or screencast 

N/A
